### ISSUE TITLE
Server crashes on startup if image type cannot be detected in an mbtiles file

### ISSUE THREAD

**Issue by brendan-ward** [consbio/mbtileserver#58]
```
Unable to create service set: could not open mbtiles file "/tmp/test.mbtiles": Could not detect tile format
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x43f7601]

goroutine 1 [running]:
github.com/consbio/mbtileserver/handlers.(*ServiceSet).Handler(0x0, 0x468d6d8, 0x1, 0xc4204121e0, 0xc4200be068)
	/Users/bcward/go/src/github.com/consbio/mbtileserver/handlers/handlers.go:405 +0x61
main.serve()
	/Users/bcward/go/src/github.com/consbio/mbtileserver/main.go:161 +0x55a
main.glob..func1(0x49203c0, 0xc420395400, 0x0, 0x4)
	/Users/bcward/go/src/github.com/consbio/mbtileserver/main.go:34 +0x20
github.com/consbio/mbtileserver/vendor/github.com/spf13/cobra.(*Command).execute(0x49203c0, 0xc4200c0100, 0x4, 0x4, 0x49203c0, 0xc4200c0100)
	/Users/bcward/go/src/github.com/consbio/mbtileserver/vendor/github.com/spf13/cobra/command.go:651 +0x23d
github.com/consbio/mbtileserver/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x49203c0, 0x400646c, 0xc42009c058, 0x0)
	/Users/bcward/go/src/github.com/consbio/mbtileserver/vendor/github.com/spf13/cobra/command.go:726 +0x2d4
github.com/consbio/mbtileserver/vendor/github.com/spf13/cobra.(*Command).Execute(0x49203c0, 0x0, 0xbed3325b39422aa3)
	/Users/bcward/go/src/github.com/consbio/mbtileserver/vendor/github.com/spf13/cobra/command.go:685 +0x2b
main.main()
	/Users/bcward/go/src/github.com/consbio/mbtileserver/main.go:66 +0x2d
```

Test file:
[test.mbtiles.zip](https://github.com/consbio/mbtileserver/files/2274574/test.mbtiles.zip)

We should not publish an invalid service,  but we also shouldn't completely crash when this occurs.

**Comment by brendan-ward** [comment]
There are other related issues that are encountered during scanning the mbtiles file, e.g., if `tiles` table is missing.  These errors should not cause a server crash.

**Comment by sikmir** [comment]
Fixed by #80

**Comment by brendan-ward** [comment]
Good catch @sikmir !
